### PR TITLE
First working version, tested on Audi A3 2008 (Engine: BMM)

### PR DESCRIPTION
**Pull request by tothpetiszilard**
If you don't mind, I did some corrections and tested the code on my A3 from 2008.
I have use the latest SDK for ESP32. My HW is a simple development board + logic level shifter (3v3 <-> 5V) + MCP2551.
Worked for me, but I did not put too much time into it. Could you do a review, please?

### CODE REVIEW

File: main/can.c [hunk 1]
```diff
@@ -61,33 +63,36 @@ void CAN_Close()
...
     return 0; //something terrible happened
 }
 
 // TODO: understand error codes for this one - 255 appears to be sucess and 0 failure.
 uint8_t CAN_SendMsg(CanMessage_t * msg){
-    can_message_t _msg;
+    twai_message_t _msg;
     _msg.identifier = msg->id;
     _msg.data_length_code = msg->len;
+    _msg.flags = 0;
```
Comment (tothpetiszilard) [inline main/can.c:86]: Flags is deprecated, also worked with `_msg.rtr = 0;` and `_msg.extd = 0;`.
I think only clearing the memory is necessary here, to remove the garbage, nothing special.

**Comment by xerootg** [comment]
So, I wrote this, well ported it anyway, and then figured out I didn't have a TP2.0 compatible ECU after all. If this works, then AWESOME. I'm guessing `log.txt` is this actually doing a thing?

**Review by xerootg — APPROVED** [review]
Merge when ready, I'm super happy to see this code going somewhere!

**Comment by tothpetiszilard** [comment]
> 

Yes, it started to work after the changes, at least with my Audi A3 8P using the OBD port. I can not merge, I don't have write access to your repo :)

**Pull request merged**